x-pack/plugins/ml/public/application/explorer/explorer_charts/explorer_charts_container_service.js

```diff
@@ -422,21 +429,57 @@ export const anomalyDataChange = function (
 function processRecordsForDisplay(anomalyRecords) {
   // Aggregate the anomaly data by detector, and entity (by/over/partition).
   if (anomalyRecords.length === 0) {
-    return [];
+    return [[], undefined];
   }
 
   // Aggregate by job, detector, and analysis fields (partition, by, over).
   const aggregatedData = {};
+
+  const jobsErrorMessage = {};
   each(anomalyRecords, (record) => {
     // Check if we can plot a chart for this record, depending on whether the source data
     // is chartable, and if model plot is enabled for the job.
     const job = mlJobService.getJob(record.job_id);
+
+    // if we already know this job has datafeed aggregations we cannot support
+    // no need to do more checks
+    if (jobsErrorMessage[record.job_id] !== undefined) {
+      return;
+    }
+
+    // check to see if datafeed has nested term aggregations with model plot disabled
+    const datafeedAggregationErrorMsg = getDatafeedAggregationsErrorMessage(job);
+    if (datafeedAggregationErrorMsg !== undefined) {
+      jobsErrorMessage[record.job_id] = datafeedAggregationErrorMsg;
+      return;
+    }
+
     let isChartable = isSourceDataChartableForDetector(job, record.detector_index);
-    if (isChartable === false && isModelPlotChartableForDetector(job, record.detector_index)) {
-      // Check if model plot is enabled for this job.
-      // Need to check the entity fields for the record in case the model plot config has a terms list.
-      const entityFields = getEntityFieldList(record);
-      isChartable = isModelPlotEnabled(job, record.detector_index, entityFields);
+    if (isChartable === false) {
+      if (isModelPlotChartableForDetector(job, record.detector_index)) {
+        // Check if model plot is enabled for this job.
+        // Need to check the entity fields for the record in case the model plot config has a terms list.
+        const entityFields = getEntityFieldList(record);
+        if (isModelPlotEnabled(job, record.detector_index, entityFields)) {
+          isChartable = true;
+        } else {
+          isChartable = false;
+          jobsErrorMessage[record.job_id] = i18n.translate(
+            'xpack.ml.timeSeriesJob.sourceDataNotChartableWithDisabledModelPlotMessage',
+            {
+              defaultMessage:
+                'source data is not viewable for this detector and model plot is disabled',
+            }
+          );
+        }
+      } else {
+        jobsErrorMessage[record.job_id] = i18n.translate(
+          'xpack.ml.timeSeriesJob.sourceDataModelPlotNotChartableMessage',
+          {
+            defaultMessage: 'both source data and model plot not chartable for this detector',
```

Comment:
I think this needs `are` inserted, `source data and model plot are not chartable`

Reply:
Updated here https://github.com/elastic/kibana/pull/84594/commits/ce964f5f6b9509419fb456c7554283d8c5a81b79